src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1848,3 +1848,16 @@ def validate_policy(namespace):
     if namespace.id is not None:
         logger.warning("\nPlease do not specify --expiry and --permissions if they are already specified in your "
                        "policy.")
+
+
+def validate_immutability_arguments(namespace):
+    from azure.cli.core.azclierror import InvalidArgumentValueError
+    if namespace.enable_alw is not None:
+        if not namespace.enable_alw:
```

Comment:
if `namespace.enable_alw` is `None`, can customer specify `--immutability-period --immutability-state --allow-append`?

Reply:
If enable_alw is None(not specified), it will create the account without the immutableStorageWithVersioning attribute (ignoring those three parameters)

Reply:
Sorry, the above behavior is because of the way I coded. Normally server throws an error.